src/Server.ts

```diff
@@ -36,7 +36,7 @@ export default class Server {
     this.router = new Router()
     this.router.post('/rpc', async ctx => {
       try {
-        const result = await this.executeEndpoint(ctx.request.body.topic, ctx.request.body.payload)
+        const result = await this.executeEndpoint(ctx.request.body.topic, ctx.request.body.payload, ctx.headers)
```

Comment:
I suggest we start the span on the receiving side (server-side). Instead of passing raw HTTP headers, we can create an `opentracing.Span` representing the client request and pass it to `executeEndpoint`.

This allows `executeEndpoint` to forward the span into the endpoint handler, allowing the endpoint handler to perform [context-aware logging](https://opentracing-javascript.surge.sh/classes/span.html#log).